homeassistant/components/nanoleaf/__init__.py

```diff
@@ -80,8 +79,13 @@ async def touch_event_callback(event: TouchEvent) -> None:
         )
     )
 
+    def _cancel_listener() -> None:
+        event_listener.cancel()
```

Comment:
Should this await the event_listener and consume `asyncio.CancelledError` in case there is an un-retrieved exception

Reply:
you mean `await event_listener.cancel()`?

Reply:
After cancelling, you can retrieve any exceptions while suppressing CancelledError and do whatever is needed with them.


```
with contextlib.suppress(asyncio.CancelledError):
   await event_listener
```

or

```
try:
  await event_listener
except Exception:
  _LOGGER.exception....
except asyncio.CancelledError:
  pass
```

Reply:
Otherwise if there is an exception in the task it will go unretrieved and log a warning when its garbage collected

Reply:
Code is already updated, Robert knew what you meant :)